sbi/src/tee_guest.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::*;
+use crate::function::*;
+
+/// Functions provided by the TEE Guest extension to TVM guests.
+#[derive(Copy, Clone, Debug)]
+pub enum TeeGuestFunction {
+    /// Marks the specified range of guest physical address space in the calling TVM as used for
+    /// emulated MMIO. The region is unpopulated; attempts to access this region will cause a
+    /// guest page fault exit with the faulting address and (a transformed version of) the faulting
+    /// instruction reported to the TVM's host, allowing the host to emulate the faulting memory
+    /// access. Both `addr` and `len` must be 4kB-aligned and must not overlap with any
+    /// previously-added regions.
+    ///
+    /// a6 = 0
+    AddEmulatedMmioRegion {
```

Comment:
Do we really need separate functions ? One functions with additional parameter to distinguish between MMIO/shared will be useful ? It will also align the TEEH function.

//Allowed only for CONFIDENTIAL MEMORY REGION
sbi_teeh_add_memory_region

//Allowed only for SHARED/MMIO region
sbi_teeg_add_memory_region

Reply:
Actually, I missed that one...it's indeed modeled as a single call for all region types (with an enum parameter) in the documentation (for the host), and as Atish noted, it should be similar when it's moved over to the guest side.

Reply:
Right, done.